### ISSUE TITLE
buy me a ☕️

### ISSUE THREAD

**Issue by alltheseas** [damus-io/damus#1596]

**Comment by robagreda** [comment]
Are we suggesting to remove the Zap button in notes? @jb55 what do you think?

**Comment by alltheseas** [comment]
> Are we suggesting to remove the Zap button in notes?

Zap button does not exist in the upcoming appstore version. 

Buy me a ☕ is an attempt to communicate to Tim Apple that with sending sats to a profile there is nothing received in return, nor is there an expectation of receiving anything.

**Comment by derekross** [comment]
I'd love to see this renamed to "Zap me a coffee". This keeps the Nostr branding of Zaps alive, allowing us to continue building the Zap brand. Zaps are important as it helps orange pill people that don't like the B word. It's something different. It has meaning. It's powerful and we should not dilute it.

**Comment by alltheseas** [comment]
Yes, we like the zaps.

The problem is that to appstore reviewers zaps may be associated with zapping a note, which is arbitrarily banned. 

"Buy me a ☕ " exists long before nostr, and Damus. "Buy me a ☕ " is common to many a projects, and equivocates with "support this creator without expecting or receiving anything in return." 

We'll discuss with @jb55 and team, and agree on way forward.

**Comment by jb55** [comment]
On Thu, Oct 12, 2023 at 09:08:22AM -0700, Daniel D’Aquino wrote:
>2. **Special note zap screen:** Some view that explains the action being performed, and implies that the zaps are only donations **to the profile/creator**, and not a "purchase" to any "digital content" in the note. Basically re-frame the note zaps as **"profile zaps done via the note"** on the UX, to increase chances that Apple would accept it, since profile zaps are allowed)
>    - Could be something similar to the mockup presented by @alltheseas if we have a "zap a coffee" feature
>    - If we don't go with the coffee idea, it could be special profile zap screen showing a profile picture, profile name, a title like "Zap this creator" with a message along the lines of "Support this creator by zapping them. Zaps are donations made voluntarily to support creators on the platform"

damus actually had this at one point, but it was still rejected because
it was initiated from a button on the post. They seem to have an issue
with the button specifically, even if its clear the button opens a
profile zap screen.

**Comment by alltheseas** [comment]
@robagreda how might a buy a ☕  flow initiation from PFP look like? 

Might we add a ☕ icon to each PFP / profile that has a LN address?

**Comment by robagreda** [comment]
> @robagreda how might a buy a ☕ flow initiation from PFP look like?
> 
> Might we add a ☕ icon to each PFP / profile that has a LN address?

I'd say that flow would be cool, easy but not sure if it's overwhelming to have a coffee mug emoji next to each PFP in the feed.

**Comment by alltheseas** [comment]
> > @robagreda how might a buy a ☕ flow initiation from PFP look like?
> > Might we add a ☕ icon to each PFP / profile that has a LN address?
> 
> I'd say that flow would be cool, easy but not sure if it's overwhelming to have a coffee mug emoji next to each PFP in the feed.

Makes sense - agree it's probably a lot of emoji for many folks. We should also keep in mind that some accessibility users disable animations to make the app usable. 

@robagreda say that Will advises buy me a ☕ / ⚡ profile initiates from PFP. How would you design to the "discoverability" feature that @danieldaquino  suggested earlier - i.e. how does the user discover the  buy me a ☕ / ⚡ profile inititation point & flow without a training course?

**Comment by robagreda** [comment]
![image](https://github.com/damus-io/damus/assets/3373850/16954d54-92cb-4af2-8bc8-c85096f24fe3)


The buy me a coffee emoji would look like this in each PFP that has enabled LN, how do you see it?

**Comment by alltheseas** [comment]
Yeah I can see it, and I notice the contrast with the top PFP, which does not have a ☕ icon

**Comment by danieldaquino** [comment]
@alltheseas, @robagreda, I like adding the little icon on the profile picture! I wonder if we should make it user configurable on whether they want to show ☕ or ⚡ under their profile picture.

> damus actually had this at one point, but it was still rejected because it was initiated from a button on the post. They seem to have an issue with the button specifically, even if its clear the button opens a profile zap screen.

@jb55, what if we add the zap button on the note, but then when the user clicks on it a message appears stating: 

> "Sorry, but zapping notes is not supported in iOS. If you would like to support this creator, please visit their profile _(Link)_ and click on the zap button. For more information, please visit this link _(Link)_". 

Do you think Apple would allow it?

@ericholguin mentioned on a previous meeting that some app does this. I think I recall seeing some messages like this on some other (non-nostr) iOS app in the past.

**Comment by derekross** [comment]
Plebstr does this and I love it because it informs the user about a missing
feature that many people are talking about across Nostr, educating them in
the process, and it explains why Damus is missing this feature, shaming
Apple.

Derek Ross

**Comment by alltheseas** [comment]
> @alltheseas, @robagreda, I like adding the little icon on the profile picture! I wonder if we should make it user configurable on whether they want to show ☕ or ⚡ under their profile picture.

Hm. I wonder if ☕ is associated with a larger minimum zap amount. I am thinking of the persona that does not appreciate being zapped fractions of a cent. 


> @jb55, what if we add the zap button on the note, but then when the user clicks on it a message appears stating: 
> 
> 
> 
> > "Sorry, but zapping notes is not supported in iOS. If you would like to support this creator, please visit their profile _(Link)_ and click on the zap button. For more information, please visit this link _(Link)_". 
> 
> 
> 
> Do you think Apple would allow it?

What's the objective with this? Might we include a link with a pre-drafted email to apple saying note zaps are p2p
like venmo, and no digital content is received in return?

**Comment by jb55** [comment]
On Mon, Oct 16, 2023 at 03:55:42PM -0700, Daniel D’Aquino wrote:
>@jb55, what if we add the zap button on the note, but then when the user clicks on it a message appears stating:
>
>> "Sorry, but zapping notes is not supported in iOS. If you would like to support this creator, please visit their profile _(Link)_ and click on the zap button. For more information, please visit this link _(Link)_".
>
>Do you think Apple would allow it?

generally no, I believe they pulled a version of a facebook app at some point for doing something like this. They don't want to be made look bad for their horrible practices and they will pull you for pointing that out.

**Comment by alltheseas** [comment]
> Plebstr does this

My impression is there is not a single, consistent standard, or perhaps incomplete reviews by different reviewers of different apps depending on how reviewer feels on that day.

> They don't want to be made look bad for their horrible practices and they will pull you for pointing that out.

Generally agree. That said, we can investigate with Vanessa from marketing perspective. @danieldaquino makes a good suggestion.

**Comment by danieldaquino** [comment]
Reposting the note that @robagreda posted on Nostr regarding some of the designs.

With regards to the votes, I counted:
- **Option 1:** 26 votes
- **Option 2:** 11 votes

I will make a first version prototype of this feature.

**Comment by danieldaquino** [comment]
@jb55, @alltheseas, I have to say I am less bullish of the idea of "buy me a coffee" than I was at the start.

What bought me into the idea initially was that I thought it would be a way to bring back note zaps and have Apple not reject it.

But if Apple would reject any zapping-related button on the post itself (even if it's rebranded as "buy me a coffee"), then I don't see why not just stick to zaps.

I think @derekross made a very good point that zaps is what makes Damus and Nostr more unique and special. If coffee won't help us bring back built-in note zaps into the app, then I don't see much benefit to it.

I will still implement a prototype with the "buy me a coffee", but I will split my work into logical pieces that will allow us to select at least two different implementations of this ticket.

**Comment by alltheseas** [comment]
When are good times for you @robagreda

**Comment by danieldaquino** [comment]
Sent a patch for the implementation of some of the changes related to discoverability. All code, testing, screenshots, and other notes can be visualized here: https://groups.google.com/a/damus.io/g/patches/c/MR2aD5P-fDw

For the coffee-specific actions I will wait until we have a meeting and figure out how to proceed with those ideas.

**Comment by alltheseas** [comment]
Closing as part of a backlog triage — we're narrowing focus to the most actionable issues to reduce noise for contributors.

If this is still relevant, please reopen with updated context or reproduction steps. Thanks for reporting!
